internal/evaluator/evaluator.go

```diff
@@ -21,12 +21,29 @@ import (
 )
 
 type Evaluator interface {
-	// TODO refactor not to expose Conftest type here
-	Evaluate(ctx context.Context, inputs []string) (CheckResults, Data, error)
+	Evaluate(ctx context.Context, inputs []string) ([]Result, Data, error)
 
 	// Destroy performs any cleanup needed
 	Destroy()
 
 	// CapabilitiesPath returns the path to the file where capabilities are defined
 	CapabilitiesPath() string
 }
+
+type Data map[string]any
+
+type Result struct {
+	FileName   string `json:"filename"`
+	Namespace  string `json:"namespace"`
+	Successes  []Rule `json:"successes,omitempty"`
+	Skipped    []Rule `json:"skipped,omitempty"`
+	Warnings   []Rule `json:"warnings,omitempty"`
+	Failures   []Rule `json:"failures,omitempty"`
+	Exceptions []Rule `json:"exceptions,omitempty"`
+}
+
+type Rule struct {
+	Message  string                 `json:"msg"`
+	Metadata map[string]interface{} `json:"metadata,omitempty"`
+	Outputs  []string               `json:"outputs,omitempty"`
```

Comment:
Are we using `Outputs`? Perhaps we can remove it?

Reply:
Probably. I'd like to do that as a follow up given the size of this PR.